src/database.rs

```diff
@@ -0,0 +1,73 @@
+use std::fs::{create_dir_all, File, OpenOptions};
+use std::io::{Result};
+use std::path::{Path, PathBuf};
+use std::{env, io};
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct RuntimeDatabase {
```

Comment:
Can we rename this `RuntimeLock`? I think we want to replace it with TCP or Unix sockets at some point. Maybe we'll retain this, maybe we won't. But I think the only use for the "runtime database" is locking.